### ISSUE TITLE
False positive of "reportIncompatibleVariableOverride" when subclassing a frozen dataclass

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#5993]
**Describe the bug**

See sample below.

Variables of a frozen dataclass are read-only, so "reportIncompatibleVariableOverride" should not be triggered.

**Code or Screenshots**

```python
# pyright: reportIncompatibleVariableOverride=true

from dataclasses import dataclass

@dataclass(frozen=True)
class C0:
    v: object

class C1(C0):
    v: int  # Reported.
```

**VS Code extension or command-line**

Pylance v2023.9.21 / Pyright 1.1.328

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.329, which I just published. It will also be included in a future release of pylance.
